src/irmin-pack/unix/pack_key_intf.ml

```diff
@@ -40,13 +48,18 @@ module type Sigs = sig
             offsets, we fetch the length information of the child at the same
             time as fetching its hash (which we must do anyway in order to do an
             integrity check), creating keys of this form. *)
-    | Indexed of 'hash
+    | Indexed : 'hash -> ('hash, safe) unsafe_state
         (** A pointer to an object in the pack file that is indexed. Reading the
             object necessitates consulting the index, after which the key can be
             promoted to {!Direct}.
 
             Such keys result from decoding pointers to other store objects
             (nodes or commits) from commits or from the branch store. *)
+    | Offset : int63 -> ('hash, unsafe) unsafe_state
+        (** Same as [Direct], but the hash and length of the object have not
+            been fetched. Only used to speed up the GC traversal. *)
+
+  type 'hash state = ('hash, safe) unsafe_state
```

Comment:
Is this mostly introduced to keep an `Offset` key from being `inspect`able?

Reply:
Yes, there is a lot of preexisting pattern matching where I would have needed to `assert false` otherwise -- but at the same time, the `node_store` is often available and as such we could imagine "promoting" the raw `Offset` to a `Direct` (but I don't see a use for `Offset` outside of the GC right now)

Reply:
Yep that makes sense. I mostly wanted to make sure I wasn't missing some type trick here. :smile:  

I agree `Offset` doesn't seem to have a usage outside of GC so it is nice to not have it in all the matches (as an aside, I've started wondering if our pervasive usage of `Pack_key.inspect` is the best pattern -- but that's for future consideration).